src/cartesi/util.lua

```diff
@@ -192,17 +192,17 @@ end
 
 local function hexhash8(hash) return string.sub(hexhash(hash), 1, 8) end
 
-local function accessdatastring(data, log2_size)
+local function accessdatastring(data, data_hash, log2_size)
     if log2_size == 3 then
         data = string.unpack("<I8", data)
         return string.format("0x%x(%u)", data, data)
     else
-        return string.format(
-            "%s...%s(2^%d bytes)",
-            hexstring(string.sub(data, 1, 3)),
-            hexstring(string.sub(data, -3, -1)),
-            log2_size
-        )
+        local data_snippet = string.format('hash:"%s"', hexhash8(data_hash))
+        if data ~= nil then
+            data_snippet =
+                string.format("%s...%s", hexstring(string.sub(data, 1, 3)), hexstring(string.sub(data, -3, -1)))
```

Comment:
No need to change, but I always forget this. I think you can use `data:sub(1,3)` instead of the `string.sub`.